### DISCUSSION TITLE
What's the difference between actions/attest-build-provenance and actions/attest-sbom?

### DISCUSSION THREAD

**Discussion by jamesbradlee** [community/community#135457] [Actions] [answered]
### Select Topic Area

Question

### Body

I'm working on a shared workflow where I want to create a build provenance, however, I'm unsure if I should run both the `actions/attest-build-provenance` action and the `actions/attest-sbom` action. What are the differences between the two, and what is the expected usage when you have one binary artifact and one bill of materials artifact?

**Comment by tinaheidinger** [comment, marked as answer]
Hi @testersen,

`actions/attest-build-provenance` and `actions/attest-sbom` are both actions designed to create attestations, but they target different types of metadata:

The `actions/attest-build-provenance` action is designed to generate and attest to the provenance of a build, including the origin and build process, to ensure integrity and reproducibility. The `actions/attest-sbom` action focuses on attesting an existing SBOM. 

Essentially, provenance attestation deals with the "how" and "where" of the software creation, while SBOM attestation addresses the "what" of its composition.

When you have a binary artifact and an SBOM artifact, you may want to use the `actions/attest-sbom` action to cryptographically attest that the SBOM is authentic and can be trusted.

**Reply by jamesbradlee** [reply]
Hey @tinaheidinger 

That is what I suspected as well - but when looking at the attestations generated by actions/attest-build-provenance and actions/attest-sbom I see no difference on the attestation page other than the created and predicate type fields, which is why I found the topic confusing.

<img width="1286" alt="image" src="https://github.com/user-attachments/assets/6ab9b435-aeb5-488f-b533-93e13c51ae4e">
<img width="1358" alt="image" src="https://github.com/user-attachments/assets/d5e09a69-8087-4a71-85a1-d7467152edf2">
<img width="1361" alt="image" src="https://github.com/user-attachments/assets/a5dbb3c1-5550-4da3-b589-edd19b1e62b7">

**Comment by RoshniRanaDS27** [comment]
hi, @testersen 

The actions/attest-build-provenance and actions/attest-sbom actions serve different purposes in your workflow, and they can be used together depending on what you want to achieve.

Differences:
actions/attest-build-provenance:

Purpose: This action generates an attestation of the build provenance, which includes details about how the artifact was built. It documents the environment, tools, dependencies, and steps involved in creating the artifact.
Usage: Use this action when you want to ensure the integrity of the build process itself, providing transparency and traceability about how a binary artifact was produced.
actions/attest-sbom:

Purpose: This action generates an attestation for a Software Bill of Materials (SBOM), which is a detailed list of components, libraries, and dependencies used within a software application. An SBOM helps in identifying vulnerabilities, licensing issues, and other security-related concerns.
Usage: Use this action when you need to attest to the components and dependencies that make up your binary artifact. It’s especially useful for maintaining a clear and secure supply chain.
Expected Usage:
Binary Artifact: For a binary artifact, you should use the actions/attest-build-provenance to document how it was built.
Bill of Materials Artifact: For the SBOM, you should use the actions/attest-sbom to attest to the list of components included in your binary.
When to Use Both:
If you have one binary artifact and one bill of materials artifact, it is beneficial to run both actions:
actions/attest-build-provenance: To attest the build process of the binary.
actions/attest-sbom: To attest the contents and dependencies listed in the SBOM.
Using both actions together provides comprehensive documentation, ensuring both the build process and the components used are properly attested, which enhances security and compliance in your workflow.

**Comment by RoshniRanaDS27** [comment]
can you also give me your point of view? for better knowledge of mine?
